src/core/router/net_db/impl.cc

```diff
@@ -978,29 +979,23 @@ std::vector<IdentHash> NetDb::GetClosestFloodfills(
   IdentHash dest_key = CreateRoutingKey(destination); {
     std::unique_lock<std::mutex> l(m_FloodfillsMutex);
     for (auto it : m_Floodfills) {
-      if (!it->IsUnreachable()) {
-        XORMetric m = dest_key ^ it->GetIdentHash();
-        if (sorted.size() < num) {
-          sorted.insert({it, m});
-        } else if (m < sorted.rbegin()->metric) {
+      if (!it->IsUnreachable() && !excluded.count(it->GetIdentHash()))
+        {
+          XORMetric m = dest_key ^ it->GetIdentHash();
           sorted.insert({it, m});
-          sorted.erase(std::prev(sorted.end()));
         }
-      }
     }
   }
+
   std::vector<IdentHash> res;
   std::uint8_t i{};
   for (auto it : sorted) {
-    if (i < num) {
-      auto& ident = it.r->GetIdentHash();
-      if (!excluded.count(ident)) {
+    while (i < num)
```

Comment:
This will insert the same hash repeatedly... and any remaining elements in the set will be ignored.

Reply:
You're right, I'll change the loop to step through the whole set.